leases/schema/mutations.py

```diff
@@ -189,6 +189,69 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class SwitchBerthMutation(graphene.ClientIDMutation):
+    class Input:
+        old_lease_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        switch_date = graphene.Date(
+            description="The date which will mark the end of the old lease and the start of the new lease."
+            "If none is provided, it will default to the day when the mutation is called"
+        )
+
+    old_berth_lease = graphene.Field(BerthLeaseNode)
+    new_berth_lease = graphene.Field(BerthLeaseNode)
+
+    @classmethod
+    @add_permission_required(BerthLease)
+    @change_permission_required(BerthLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **input):
+        old_lease_id = input.pop("old_lease_id")
+        new_berth_id = input.pop("new_berth_id")
+        switch_date = input.get("switch_date") or calculate_berth_lease_start_date()
+
+        old_lease = get_node_from_global_id(
+            info, old_lease_id, only_type=BerthLeaseNode, nullable=False,
+        )
+
+        try:
+            contract = old_lease.contract
+        except ObjectDoesNotExist as e:
+            # Flatten all the error messages on a single list
+            raise VenepaikkaGraphQLError(str(e))
+
+        kwargs = {
+            "boat": old_lease.boat,
+            "customer": old_lease.customer,
+            "end_date": old_lease.end_date,
+            "renew_automatically": old_lease.renew_automatically,
+            "status": old_lease.status,
+        }
+
+        old_lease = terminate_lease(
+            info, old_lease_id, BerthLeaseNode, end_date=switch_date,
+        )
+        old_lease.comment = "Lease is terminated due to berth switch"
+        old_lease.save(update_fields=["comment"])
+
+        new_berth = get_node_from_global_id(
+            info, new_berth_id, only_type=BerthNode, nullable=False,
+        )
```

Comment:
Can we get the berth before terminating the lease? in case it would fail or something, so the lease won't be terminated for nothing.

Reply:
Yea. Now that I think about it, should we also put the whole process in a transaction?